lucet-runtime/lucet-runtime-internals/src/instance.rs

```diff
@@ -533,10 +533,15 @@ impl Instance {
     /// modified by this call; it is the embedder's responsibility to clear or reset their state if
     /// necessary.
     ///
+    /// This will also reinitialize the kill state, which means that any outstanding
+    /// [`KillSwitch`](struct.KillSwitch.html) objects will be unable to terminate this instance.
+    /// It is the embedder's responsibility to initialize new killswitches after resetting an
+    /// instance.
```

Comment:
[This change](https://github.com/bytecodealliance/lucet/pull/429/files#diff-a99c9cb71117942c648b057de7d9a29aR562) is responsible for this, but I wanted to highlight this as a change that I feel _weird_ about. I think there's a better way forward that I'm just not seeing.

Reply:
https://github.com/bytecodealliance/lucet/pull/429#discussion_r381623306

Discussion about this continues over here. Resolving this.